pkg/webhook/controlplane/genericmutator/mutator.go

```diff
@@ -102,6 +108,39 @@ func (m *mutator) InjectClient(client client.Client) error {
 	return nil
 }
 
+type ensurerContext struct {
+	client  client.Client
+	object  metav1.Object
+	cluster *extensionscontroller.Cluster
+}
+
+// NewEnsurerContext creates a ensurer context object.
+func NewEnsurerContext(client client.Client, object metav1.Object) EnsurerContext {
+	return &ensurerContext{
+		client: client,
+		object: object,
+	}
+}
+
+// NewInternalEnsurerContext creates a ensurer context object.
+func NewInternalEnsurerContext(cluster *extensionscontroller.Cluster) EnsurerContext {
```

Comment:
Why do we need this function? Also, why is it called "internal"? What shall this mean?

Reply:
Well, this is primarily needed for the tests.